### PR TITLE
Issue reporting system  for simulator

### PR DESCRIPTION
**Pull request by yash15601**
Fixes  https://github.com/CircuitVerse/CircuitVerse/issues/1158

#### Describe the changes you have made in this PR -
Added a small sidebar in simulator  for reporting issues which will report issues in slack channel

### Screenshots of the changes (If any) -
image

image

image

image

### CODE REVIEW

**Comment by yash15601** [comment]
@nitin10s  @tachyons  @satu0king  Review please

File: app/assets/stylesheets/simulator.scss [hunk 1]
```diff
@@ -15,3 +15,21 @@
   overflow-x: hidden;
   overflow-y: scroll;
 }
+
+#mySidebar a{
+  position: absolute;
```
[the remark this answers is not in the document]
Comment (satu0king) [reply app/assets/stylesheets/simulator.scss]: Please don't close these codeclimate recommendations, at least try fixing them

File: app/views/simulator/edit.html.erb [hunk 2]
```diff
@@ -297,4 +297,46 @@
       <script src="/js/Quin_McCluskey.js"></script>
     <% end %>
     <script src="/js/canvas2svg.js"></script>
-    <div>
+    
+    <!---issue reporting-system----->
+    <div class="mysidebar">
```
Comment (satu0king) [inline app/views/simulator/edit.html.erb]: `mysidebar` is misleading, we already have a `sidebar`. Please use names like `report-sidebar` or `feedback-sidebar`

**Comment by satu0king** [comment]
> Please add a close button

Yes please add but not completely necessary as user can exit by clicking anywhere else. But *you have successfully submitted should come* only after clicking report 🙄. Otherwise Form is aesthetically pleasing

File: app/views/simulator/edit.html.erb (hunk 2 above)
Comment (yash15601) [reply app/views/simulator/edit.html.erb]: will change this soon

**Comment by yash15601** [comment]
@satu0king  okay will try for that will fix soon

**Comment by satu0king** [comment]
I think its important that we add other details like `user_id` and `current link` to the text as well. Even if we don't capture screenshot, we should be able to capture this.

**Comment by yash15601** [comment]
@satu0king  current link ?

**Comment by yash15601** [comment]
@nitin10s  @satu0king  done Review please

**Comment by satu0king** [comment]
> @satu0king current link ?

Suppose feedback is "and gate not working correctly"

I want it to be "and gate not working correctly. URL: https://circuitverse.org/simulator/edit/66, User ID:3, User Name: Satvik Ramaprasasd"

**Comment by yash15601** [comment]
@satu0king  okay we can try this

**Comment by yash15601** [comment]
@nitin10s  review

**Pull request closed without merging**